wfe/web-front-end_test.go

```diff
@@ -308,9 +308,15 @@ func TestHandleFunc(t *testing.T) {
 
 	// Disallowed method special case: response to HEAD has got no body
 	runWrappedHandler(&http.Request{Method: "HEAD"}, "GET", "POST")
-	test.AssertEquals(t, stubCalled, false)
+	test.AssertEquals(t, stubCalled, true)
 	test.AssertEquals(t, rw.Body.String(), "")
-	test.AssertEquals(t, sortHeader(rw.Header().Get("Allow")), "GET, POST")
+
+	// HEAD doesn't work with POST-only endpoints
+	runWrappedHandler(&http.Request{Method: "HEAD"}, "POST")
+	test.AssertEquals(t, stubCalled, false)
+	test.AssertEquals(t, rw.Header().Get("Content-Type"), "application/problem+json")
+	test.AssertEquals(t, rw.Body.String(), `{"type":"urn:acme:error:malformed","detail":"Method not allowed"}`)
```

Comment:
@jsha "erroneously sends a body in response to a HEAD when methodsOK does not include GET" is right here: method was `"HEAD"` but we got a non-empty body. I fixed the bug and removed this line while merging into #788 but I should probably put it back as `test.AssertEquals(t, rw.Body.String(), "")`